temporal-serviceclient/src/main/java/io/temporal/internal/retryer/GrpcRetryer.java

```diff
@@ -32,41 +32,69 @@
 
 public final class GrpcRetryer {
 
-  private final Supplier<GetSystemInfoResponse.Capabilities> serverCapabilities;
-
+  @FunctionalInterface
   public interface RetryableProc<E extends Throwable> {
     void apply() throws E;
+
+    default RetryableFunc<Void, E> asFunc() {
+      return () -> {
+        RetryableProc.this.apply();
+        return null;
+      };
+    }
   }
 
+  @FunctionalInterface
   public interface RetryableFunc<R, E extends Throwable> {
     R apply() throws E;
   }
 
-  public GrpcRetryer(Supplier<GetSystemInfoResponse.Capabilities> serverCapabilities) {
+  public static <T extends Throwable> void retry(
```

Comment:
I'd prefer not to change this if there is not a user benefit.

Reply:
I realize the benefit is tiny, but creating less garbage on the GC heap is a user benefit.

Reply:
Not worth it in my opinion, if we every do need to add any state to the retrier will have to revert this change anyway.